airflow/config_templates/config.yml

```diff
@@ -366,6 +366,17 @@
       type: string
       example: "path.to.CustomXCom"
       default: "airflow.models.xcom.BaseXCom"
+    - name: user_defined_execute_context
```

Comment:
I am not sure if the user should be able to overwrite this context manager. Can you provide use cases when necessary? 

I recently thought a lot about a similar mechanism to be able to define a context that will allow authorization to [gcloud](https://github.com/apache/airflow/pull/8432), or [forward ports via SSH](https://github.com/apache/airflow/issues/8615). However, I am not convinced that this should be defined at the global level.

Reply:
See my explanation below, I think the name "user_defined_execute_context" has misled you :)